test/deployconfig.jl

```diff
@@ -260,20 +269,54 @@ end end
                                      devbranch="master", devurl="hello-world", push_preview=true,
                                      repo_previews="github.com/JuliaDocs/Documenter-previews.jl.git",
-                                     branch_previews="gh-pages-previews")
+                                     branch_previews="gh-pages-previews", deploy_on_schedule=false)
         @test d.all_ok
         @test d.subfolder == "previews/PR42"
         @test d.repo == "github.com/JuliaDocs/Documenter-previews.jl.git"
         @test d.branch == "gh-pages-previews"
         d = Documenter.deploy_folder(cfg; repo="github.com/JuliaDocs/Documenter.jl.git",
                                      devbranch="not-master", devurl="hello-world", push_preview=false,
                                      repo_previews="",
-                                     branch_previews="")
+                                     branch_previews="", deploy_on_schedule=false)
         @test !d.all_ok
         @test Documenter.authentication_method(cfg) === Documenter.SSH
         @test Documenter.documenter_key(cfg) === "SGVsbG8sIHdvcmxkLg=="
         @test Documenter.documenter_key_previews(cfg) === "SGVsbG8sIHdvcmxkLw=="
     end
+    # Build on `schedule`
+    withenv("GITHUB_EVENT_NAME" => "schedule",
+            "GITHUB_REPOSITORY" => "JuliaDocs/Documenter.jl",
+            "GITHUB_REF" => "refs/tags/v1.2.3",
+            "GITHUB_ACTOR" => "github-actions",
+            "GITHUB_TOKEN" => "SGVsbG8sIHdvcmxkLg==",
+            "DOCUMENTER_KEY" => nothing,
+        ) do
+        cfg = Documenter.GitHubActions()
+        d = Documenter.deploy_folder(cfg; repo="github.com/JuliaDocs/Documenter.jl.git",
+                                     devbranch="master", devurl="dev", push_preview=true,
+                                     deploy_on_schedule=true)
+        @test d.all_ok
+        @test d.subfolder == "v1.2.3"
+        @test d.repo == "github.com/JuliaDocs/Documenter.jl.git"
+        @test d.branch == "gh-pages"
+        @test Documenter.authentication_method(cfg) === Documenter.HTTPS
+        @test Documenter.authenticated_repo_url(cfg) === "https://github-actions:SGVsbG8sIHdvcmxkLg==@github.com/JuliaDocs/Documenter.jl.git"
+    end
+
+    # Don't build on `schedule` if `deploy_on_schedule`=false
+    withenv("GITHUB_EVENT_NAME" => "schedule",
+        "GITHUB_REPOSITORY" => "JuliaDocs/Documenter.jl",
+        "GITHUB_REF" => "refs/tags/v1.2.3",
+        "GITHUB_ACTOR" => "github-actions",
+        "GITHUB_TOKEN" => "SGVsbG8sIHdvcmxkLg==",
+        "DOCUMENTER_KEY" => nothing,
+        ) do
+        cfg = Documenter.GitHubActions()
+        d = Documenter.deploy_folder(cfg; repo="github.com/JuliaDocs/Documenter.jl.git",
+                                        devbranch="master", devurl="dev", push_preview=true,
+                                        deploy_on_schedule=false)
+        @test !d.all_ok
```

Comment:
Might be better to explicitly test that this failed because the event name `schedule` isn't accepted when `deploy_on_schedule=false`.